modules/claims_api/spec/requests/v2/veteran_identifier_request_spec.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Veteran Identifier Endpoint', type: :request do
+  let(:path) { '/services/benefits/v2/veteran-id' }
+  let(:headers) { { 'Authorization': 'Bearer somerandomstuff' } }
+  let(:data) { File.read(Rails.root.join('modules', 'claims_api', 'spec', 'fixtures', 'veteran_identifier.json')) }
+
+  context 'when auth header and body params are present' do
+    describe 'veteran identifier' do
+      it 'returns an id' do
+        post path, params: data, headers: headers
+        icn = JSON.parse(response.body)['id']
+        expect(icn).to eq(ClaimsApi::V2::VeteranIdentifierController::ICN_FOR_TEST_USER)
+        expect(response.status).to eq(200)
+      end
+    end
+  end
+
+  context 'when body params are not present' do
+    let(:data) { nil }
+
+    describe 'veteran identifier' do
+      it 'returns a 422 error code' do
```

Comment:
shouldnt this be a 400 in this case?

Reply:
JSON schema validation will raise an Unprocessable Entity exception which will return a 422 if any of the required params are missing.

Reply:
sorry, what i meant was not what is currently implemented but what is expected, and a 400 for invalid input is more what i expect from a rest api. it is a "bad request"